tests/unit_tests/test_cli.py

```diff
@@ -213,26 +213,99 @@ def test_invalid_config_via_env(runner: CliRunner):
 
 @responses.activate
 def test_submit_plan(runner: CliRunner):
-    body_data = {"name": "sleep", "params": {"time": 5}}
+    body_data = {
+        "name": "sleep",
+        "params": {"time": 5},
+        "instrument_session": "cm12345-1",
+    }
 
     response = responses.post(
         url="http://a.fake.host:12345/tasks",
         match=[matchers.json_params_matcher(body_data)],
     )
 
     config_path = "tests/unit_tests/example_yaml/rest_and_stomp_config.yaml"
+    output = runner.invoke(
+        main,
+        [
+            "-c",
+            config_path,
+            "controller",
+            "run",
+            "-i",
+            "cm12345-1",
+            "sleep",
+            '{"time": 5}',
+        ],
+    )
+
+    assert response.call_count == 1, output.output
+
+
+@patch("blueapi.cli.cli.DiskCache")
+@responses.activate
+def test_submit_plan_with_cached_instrument_session(
+    mock_cache: Mock, runner: CliRunner
+):
+    body_data = {
+        "name": "sleep",
+        "params": {"time": 5},
+        "instrument_session": "cm12345-1",
+    }
+
+    response = responses.post(
+        url="http://a.fake.host:12345/tasks",
+        match=[matchers.json_params_matcher(body_data)],
+    )
+
     runner.invoke(
-        main, ["-c", config_path, "controller", "run", "sleep", '{"time": 5}']
+        main,
+        ["controller", "set-instrument-session", "cm12345-1"],
     )
+    config_path = "tests/unit_tests/example_yaml/rest_and_stomp_config.yaml"
 
-    assert response.call_count == 1
+    mock_cache.return_value = Mock()
+    mock_cache.return_value.get.return_value = "cm12345-1"
+    output = runner.invoke(
+        main,
+        [
+            "-c",
+            config_path,
+            "controller",
+            "run",
+            "sleep",
+            '{"time": 5}',
+        ],
+    )
+
+    assert response.call_count == 1, output.output
+
+
+@patch("blueapi.cli.cli.DiskCache")
+def test_caches_instrument_session(_: Mock, runner: CliRunner):
+    output = runner.invoke(
+        main,
+        ["controller", "set-instrument-session", "cm12345-1"],
+    )
+
+    assert output.stdout == "Default instrument session set to cm12345-1\n"
```

Comment:
Probably also worth asserting that it was set

```
mock_cache.set.assert_called_once_with("instrument_session", "cm12345-1")
```

(or whatever the called_once_with method should be called)